ui/imports/shared/stores/PermissionsStore.qml

```diff
@@ -11,15 +11,8 @@ QtObject {
 
     required property string activeSectionId
     required property string activeChannelId
-    required property var chatCommunitySectionModuleInst
-
-    // all permissions model
-    readonly property var permissionsModel:
-        chatCommunitySectionModuleInst.permissionsModel
-
-    function setHideIfPermissionsNotMet(chatId, checked) {
-        //TODO: backend implementation
-    }
+    required property bool allTokenRequirementsMet
+    required property var permissionsModel
```

Comment:
I'm not big fan of changing the api that way, but after closer look I can't invent better solution... However I would recommend clearly divide API into two sections.

First, major section would contain the API that is needed by UI:
- read only properties (even if some of them are not marked as read only bc they are also used to pass data from store to store internally)
- methods / signals

Second section would contain any other additional stuff not needed by UI but exposed because used by stores itself (and present there only because of limitations of QML). Especially for signals `createOrEditCommunityTokenPermission` it should be used that they are for internal use in stores, not part of the API for UI.